### ISSUE TITLE
[Idea] Crawl variables from a local configuration file

### ISSUE THREAD

**Issue by Xylemon** [GameServerManagers/LinuxGSM#344]
While writing some shell scripts of my own, I just got an idea that might make things easier. Since you have to go in and change certain variables and redownload the whole script if a big update happens, why not make certain things have $variables and you can edit a local configuration file without fooling around with the scripts?

**Comment by dgibbs64** [comment]
There may be a feature being added soon that does a similar thing do ti with muiiple configs and servers

**Comment by Xylemon** [comment]
Did anything ever come out of this? I've been putting off updating my LGSM scripts because I host several servers and it's honestly a waste of time git cloning the scripts then going through and editing the variables. It would be incredibly easy to implement  inside the server scripts a command that loads a config file that comes with the script and then the server variables are stored inside the config file and the script reads from the variables from the file. I have already done something like this in a series of bash scripts I wrote: 

Loads the config file
https://github.com/Xylemon/Window-Manager-Scripts/blob/master/videobackground#L11

The config file itself
https://github.com/Xylemon/Window-Manager-Scripts/blob/master/.xyscripts

I would totally be up to forking and making a working prototype if interested, it just would make life a 1000 times easier.

**Comment by dgibbs64** [comment]
Yeah it is planned to have separate config files for varfiables. However certain things need to be in place before this is done. The current development branch has several significant core updates and improvements to make this type of feature workable. But yeah it is something I do plan on implementing.

**Comment by Xylemon** [comment]
@dgibbs64 I'm glad to hear this is still planned, however do you mind me asking what needs to be done before this is implemented? The way I had in mind when it comes to implementing this would be really simple, unless the way I was going to do it wouldn't work with the scripts.

**Comment by dgibbs64** [comment]
There are just in general a few different significant changes that require a lot of thought and planning. There have been changes to the directory structure and the download system most notably. 

But the idea it similar to how yours works with the main and complex area being managing multiple server instances. 

As you can probably tell there is a large backlog of stuff for me to work on currently but when the time is right there will be big changes to how to main file e.g csgoserver works and will be separated in to its own configs

**Comment by Xylemon** [comment]
Sounds good, I'll keep an eye on updates.

**Comment by dgibbs64** [comment]
There has been a major update to better address config files and multi server management.
